subsys/net/ip/tcp2_priv.h

```diff
@@ -169,6 +169,10 @@ struct tcp { /* TCP connection */
 	struct tcp_options recv_options;
 	struct k_delayed_work send_timer;
 	sys_slist_t send_queue;
+	struct k_delayed_work send_data_timer;
+	struct net_pkt *send_data;
+	size_t send_data_retries;
```

Comment:
Would u8_t be enough here, after all we probably will not retransmit very long.

Reply:
Sure, fixed.